### PR TITLE
tests: performance-metrics: Add timeout parameter as runtime argument.

### PR DESCRIPTION
**Pull request by smit-gardhariya**
This commit will add timeout parameter to performance metrics testcases as runtime argument. Testcases are facing the timeout issue sometimes when we run it with MSHV hypervisor. This change will provide mechanism to pass the timeout as per the need while running the testcases. If nothing is passed, default timeout or timeout set as per the testcase will be followed.

### CODE REVIEW

**Review by rbradford — COMMENTED** [review]
Looks okay to me but @likebreath is the expert on this code.

**Review by likebreath — COMMENTED** [review]
@smit-gardhariya  Thank you for the contribution. Just want to remind you that you can always increase the time-out for certain tests if mshv needs more time for these tests. Using a global option that overrides all tests at once seems to be fairly inflexible to me. 

With that being said, the changes look fine to me.  I will let you decide which way is more suitable for your use case. Just let me know.

**Comment by smit-gardhariya** [comment]
> 

@likebreath Thank you for the quick review. With MSHV, testcases are a bit flaky sometimes w.r.t timeout. So increasing timeout for the testcase may require change again in future. I would like to go with these changes which will let user run performance metrics testcases with desired timeout so we can get the result of perf metrics.
Please let me know your thoughts on this.

@jinankjain Could you please suggest here.

**Review by likebreath — APPROVED** [review]

**Comment by likebreath** [comment]
> With MSHV, testcases are a bit flaky sometimes w.r.t timeout. So increasing timeout for the testcase may require change again in future. I would like to go with these changes which will let user run performance metrics testcases with desired timeout so we can get the result of perf metrics. Please let me know your thoughts on this.

Sounds good to me. Just approved the PR. 

@smit-gardhariya Let's wait for one more approval from the mshv team before land it. Please help ping whoever you find suitable. Thank you.

**Review by jinankjain — APPROVED** [review]

**Pull request merged**
